### DISCUSSION TITLE
What is the .directory entry on .gitignore?

### DISCUSSION THREAD

**Discussion by kmk3** [netblue30/firejail#5701] [Q&A] [answered]
I am working on some improvements related to .gitignore and I noticed that
there is a `.directory` entry, which was added on commit 11d9e0a7c ("update
gitignore", 2019-08-07) by @smitsohu.

What exactly is that?

What kind of application creates it?

The only instance of `.directory` that I found is the following, which I'm not
sure if it's related to the one on .gitignore:

From <https://wiki.archlinux.org/title/Desktop_entries>:

> Directory
> 
> Defines the appearance of a submenu in the application menu. Directory
> entries use the .directory file extension.

Which also seems unclear to me (why is it not just a normal directory?).

**Comment by ghost** [comment, marked as answer]
The .directory entry on .gitignore might be different from the [desktop-entry-spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming):

> Desktop entry files should have the .desktop extension, except for files of Type Directory which should have the .directory extension.

I only have one such .directory file on my machine, in a stale folder for a claws-mail breeze theme:

```
[Dolphin]
SortRole=type
Timestamp=2016,11,16,15,45,28
Version=3
ViewMode=1
VisibleRoles=Details_text,Details_size,Details_date,Details_type,CustomizedDetails

[Settings]
HiddenFilesShown=true
```

Just a FYI, not sure what it does on .gitignore...

**Reply by kmk3** [reply]
I see; thanks for the example.

**Reply by smitsohu** [reply]
My .directory files are similar to @glitsj16's. I think they enable customization of directory appearance in KDE file managers and the KDE file selector.

Some tools create these files automatically, which probably is the reason why I added it to the .gitignore.

**Reply by kmk3** [reply]
> My .directory files are similar to @glitsj16's. I think they enable
> customization of directory appearance in KDE file managers and the KDE file
> selector.
> 
> Some tools create these files automatically, which probably is the reason why
> I added it to the .gitignore.

That makes sense, thanks.

**Reply by kmk3** [reply]
Oh I remember it now that you mentioned KDE (and looking at the example again):
.directory is the equivalent to desktop.ini on Windows.

That is, it stores file manager settings per directory (such as show list vs
icon view, thumbnail size, show/hide hidden files).

Dolphin would by default create these files everywhere, but then they changed
it to stop doing that.
